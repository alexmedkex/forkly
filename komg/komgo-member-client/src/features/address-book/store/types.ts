import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
