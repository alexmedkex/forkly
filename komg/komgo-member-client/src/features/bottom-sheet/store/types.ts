import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
