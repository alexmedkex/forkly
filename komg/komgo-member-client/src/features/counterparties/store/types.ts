import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import { IX500Name } from '@komgo/types'

// Actions types
export enum CounterpartiesActionType {
  FETCH_CONNECTED_COUNTERPARTIES_REQUEST = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_REQUEST',
  FETCH_CONNECTED_COUNTERPARTIES_SUCCESS = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_SUCCESS',
  FETCH_CONNECTED_COUNTERPARTIES_FAILURE = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_FAILURE',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_REQUEST = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_REQUEST',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE',
  ADD_COUNTERPARTY_REQUEST = '@@counterparties/ADD_COUNTERPARTY_REQUEST',
  ADD_COUNTERPARTY_SUCCESS = '@@counterparties/ADD_COUNTERPARTY_SUCCESS',
  ADD_COUNTERPARTY_FAILURE = '@@counterparties/ADD_COUNTERPARTY_FAILURE',
  FETCH_COUNTERPARTY_REQ_REQUEST = '@@counterparties/FETCH_COUNTERPARTY_REQ_REQUEST',
  FETCH_COUNTERPARTY_REQ_SUCCESS = '@@counterparties/FETCH_COUNTERPARTY_REQ_SUCCESS',
  FETCH_COUNTERPARTY_REQ_FAILURE = '@@counterparties/FETCH_COUNTERPARTY_REQ_FAILURE',
  RESEND_COUNTERPARTY_REQUEST = '@@counterparties/RESEND_COUNTERPARTY_REQUEST',
  RESEND_COUNTERPARTY_SUCCESS = '@@counterparties/RESEND_COUNTERPARTY_SUCCESS',
  RESEND_COUNTERPARTY_FAILURE = '@@counterparties/RESEND_COUNTERPARTY_FAILURE',
  RESPONSE_ON_COUNTERPARTY_REQ_REQUEST = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_REQUEST',
  RESPONSE_ON_COUNTERPARTY_REQ_SUCCESS = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_SUCCESS',
  RESPONSE_ON_COUNTERPARTY_REQ_FAILURE = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_FAILURE',
  SEARCH_COUNTERPARTY = '@@counterparties/SEARCH_COUNTERPARTY',
  SORT_CONNECTED_COUNTERPARTIES = '@@counterparties/SORT_CONNECTED_COUNTERPARTIES',
  SET_ADD_COUNTERPARTIES = '@@counterparties/SET_ADD_COUNTERPARTIES',
  SET_COUNTERPARTY_MODAL = '@@counterparties/SET_COUNTERPARTY_MODAL',
  SET_REQUEST_ACTION = '@@counterparties/SET_REQUEST_ACTION',
  FETCH_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_REQUEST',
  FETCH_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_SUCCESS',
  FETCH_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_FAILURE',
  UPDATE_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_REQUEST',
  UPDATE_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_SUCCESS',
  UPDATE_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_FAILURE',
  CREATE_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_REQUEST',
  CREATE_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_SUCCESS',
  CREATE_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_FAILURE'
}

// State
export interface CounterpartiesStateFields {
  counterpartiesSearch: string
  counterparties: Counterparty[]
  counterpartiesSort: Sort
  notConnectedCounterpartySearch: string
  notConnectedCounterparties: NotConnectedCounterparty[]
  addCounterparties: string[]
  isAddModalOpen: boolean
  requestResponseActionStatus: boolean
  counterpartyRequest: CounterpartyRequest | null
  counterpartyProfiles: Map<string, CounterpartyProfile>
}

// Immutable state
export type CounterpartiesState = ImmutableMap<CounterpartiesStateFields>

export enum CouneterpartyStatus {
  COMPLETED = 'COMPLETED',
  PENDING = 'PENDING',
  WAITING = 'WAITING'
}

export const CouneterpartyStatusText: { [key: string]: string } = {
  [CouneterpartyStatus.COMPLETED]: 'Completed',
  [CouneterpartyStatus.PENDING]: 'Request pending',
  [CouneterpartyStatus.WAITING]: 'Action Required'
}

// Data types
export interface Counterparty {
  isFinancialInstitution: boolean
  staticId: string
  isMember: boolean
  x500Name: IX500Name
  covered: boolean
  status?: string
  timestamp?: Date
  profile?: CounterpartyProfile
}

export interface CounterpartyRequest extends Counterparty {
  requestId: string
}

export interface CompanyInfo {
  CN: string
  O: string
  C: string
  L: string
  STREET: string
  PC: string
}

export interface NotConnectedCounterparty {
  staticId: string
  isMember: boolean
  x500Name: CompanyInfo
  status?: CouneterpartyStatus
}

export interface SearchCounterpartyPayload {
  search: string
  typeCounterparty: 'counterpartiesSearch' | 'notConnectedCounterpartySearch'
}

export interface Sort {
  column: string
  order: 'ascending' | 'descending' | ''
}

export interface ModalPayload {
  name: 'isAddModalOpen'
  value: boolean
}

export interface CounterpartyProfile {
  id: string
  counterpartyId: string
  riskLevel: RiskLevel
  renewalDate: string
  managedById: string
}

export enum RiskLevel {
  unspecified = '',
  low = 'low',
  medium = 'medium',
  high = 'high'
}

// Actions
export interface FetchCounterpartiesSuccess extends Action {
  payload: Counterparty[]
  type: CounterpartiesActionType.FETCH_CONNECTED_COUNTERPARTIES_SUCCESS
}

export interface FetchCounterpartiesError extends Action {
  type: CounterpartiesActionType.FETCH_CONNECTED_COUNTERPARTIES_FAILURE
  payload: string
}

export interface FetchNotConnectedCounterpartiesSuccess extends Action {
  payload: NotConnectedCounterparty[]
  type: CounterpartiesActionType.FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS
}

export interface FetchNotConnectedCounterpartiesError extends Action {
  type: CounterpartiesActionType.FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE
  payload: string
}

export interface RequestFetchSuccess extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_REQ_SUCCESS
  payload: any
}

export interface SearchCounterparty extends Action {
  type: CounterpartiesActionType.SEARCH_COUNTERPARTY
  payload: SearchCounterpartyPayload
}

export interface SortConnectedCounterparties extends Action {
  type: CounterpartiesActionType.SORT_CONNECTED_COUNTERPARTIES
  payload: Sort
}

export interface AddCounterpartySuccess extends Action {
  type: CounterpartiesActionType.ADD_COUNTERPARTY_SUCCESS
  payload: string[]
}

export interface SetCounterpartyModal extends Action {
  type: CounterpartiesActionType.SET_COUNTERPARTY_MODAL
  payload: ModalPayload
}

export interface SetAddCounterparties extends Action {
  type: CounterpartiesActionType.SET_ADD_COUNTERPARTIES
  payload: string[]
}

export interface CounterPartyResponseAction extends Action {
  type: CounterpartiesActionType.SET_REQUEST_ACTION
  payload: {
    status: boolean
  }
}

export interface FetchCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

export interface FetchCounterpartyProfileFailure extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_PROFILE_FAILURE
  payload: {
    error: any
    counterpartyId: string
  }
}

export interface CreateCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.CREATE_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

export interface UpdateCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.UPDATE_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

// All actions
export type CounterpartiesAction =
  | FetchCounterpartiesSuccess
  | FetchCounterpartiesError
  | RequestFetchSuccess
  | SetCounterpartyModal
  | FetchNotConnectedCounterpartiesError
  | FetchNotConnectedCounterpartiesSuccess
  | SearchCounterparty
  | SortConnectedCounterparties
  | AddCounterpartySuccess
  | SetAddCounterparties
  | CounterPartyResponseAction
  | FetchCounterpartyProfileSuccess
  | FetchCounterpartyProfileFailure
  | CreateCounterpartyProfileSuccess
  | UpdateCounterpartyProfileSuccess
