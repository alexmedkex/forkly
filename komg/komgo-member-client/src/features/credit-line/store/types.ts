import { ImmutableMap } from '../../../utils/types'
import { Map } from 'immutable'
import {
  ICreditLine,
  IRiskCoverData,
  Currency,
  ICreditLineResponse,
  ISharedCreditLine,
  IProductContext,
  ICreditLineRequest,
  ICreateCreditLineRequest,
  IInformationShared
} from '@komgo/types'
import { Products } from '../../document-management/constants/Products'
import { SubProducts } from '../../document-management/constants/SubProducts'
import { IMember } from '../../members/store/types'

export enum CreditLineActionType {
  FetchCreditLinesRequest = '@@risk-cover/FETCH_CREDIT_LINES_REQUEST',
  FetchCreditLinesSuccess = '@@risk-cover/FETCH_CREDIT_LINES_SUCCESS',
  FetchCreditLinesFailure = '@@risk-cover/FETCH_CREDIT_LINES_FAILURE',
  GetCreditLineRequest = '@@risk-cover/GET_CREDIT_LINE_REQUEST',
  GetCreditLineSuccess = '@@risk-cover/GET_CREDIT_LINE_SUCCESS',
  GetCreditLineFailure = '@@risk-cover/GET_CREDIT_LINE_FAILURE',
  CreateCreditLineRequest = '@@risk-cover/CREATE_CREDIT_LINE_REQUEST',
  CreateCreditLineSuccess = '@@risk-cover/CREATE_CREDIT_LINE_SUCCESS',
  CreateCreditLineFailure = '@@risk-cover/CREATE_CREDIT_LINE_FAILURE',
  EditCreditLineRequest = '@@risk-cover/EDIT_CREDIT_LINE_REQUEST',
  EditCreditLineSuccess = '@@risk-cover/EDIT_CREDIT_LINE_SUCCESS',
  EditCreditLineFailure = '@@risk-cover/EDIT_CREDIT_LINE_FAILURE',
  FetchDisclosedCreditLineSummariesRequest = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_REQUEST',
  FetchDisclosedCreditLineSummariesSuccess = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_SUCCESS',
  FetchDisclosedCreditLineSummariesFailure = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_FAILURE',
  FetchDisclosedCreditLinesForCounterpartyRequest = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_REQUEST',
  FetchDisclosedCreditLinesForCounterpartySuccess = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_SUCCESS',
  FetchDisclosedCreditLinesForCounterpartyFailure = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_FAILURE',
  RemoveCreditLineRequest = '@@risk-cover/REMOVE_CREDIT_LINE_REQUEST',
  RemoveCreditLineSuccess = '@@risk-cover/REMOVE_CREDIT_LINE_SUCCESS',
  RemoveCreditLineFailure = '@@risk-cover/REMOVE_CREDIT_LINE_FAILURE',
  CreateReqInformationRequest = '@@risk-cover/CREATE_REQ_INFORMATION_REQUEST',
  CreateReqInformationSuccess = '@@risk-cover/CREATE_REQ_INFORMATION_SUCCESS',
  CreateReqInformationFailure = '@@risk-cover/CREATE_REQ_INFORMATION_FAILURE',
  FetchRequestsRequest = '@@risk-cover/FETCH_REQ_REQUEST',
  FetchRequestsSuccess = '@@risk-cover/FETCH_REQ_SUCCESS',
  FetchRequestsFailure = '@@risk-cover/FETCH_REQ_FAILURE',
  DeclineAllRequestsRequest = '@@risk-cover/DECLINE_ALL_REQ_REQUEST',
  DeclineAllRequestsSuccess = '@@risk-cover/DECLINE_ALL_REQ_SUCCESS',
  DeclineAllRequestsFailure = '@@risk-cover/DECLINE_ALL_REQ_FAILURE'
}

export enum CreditLineType {
  RiskCover = 'riskCover',
  BankLine = 'bankLine'
}

export interface IExtendedCreditLine extends ICreditLineResponse {
  counterpartyName: string
  counterpartyLocation: string
  data: IRiskCoverData
  sharedCreditLines: IExtendedSharedCreditLine[]
}

export interface IExtendedSharedCreditLine extends ISharedCreditLine<IInformationShared> {
  updatedAt?: string
  counterpartyName: string
}

export interface ICreateOrEditSharedCreditLine {
  counterpartyStaticId: string
  sharedWithStaticId: string
  data: IInformationShared
  requestStaticId?: string
  staticId?: string
  updatedAt?: string
}
export interface ICreateOrEditCreditLineForm {
  context: IProductContext
  counterpartyStaticId: string
  appetite: boolean
  availability: boolean
  creditLimit?: number
  availabilityAmount?: number
  data?: IRiskCoverData
  currency: Currency
  sharedCreditLines: ICreateOrEditSharedCreditLine[]
  availabilityAmountUpdatedAt?: string | Date
  staticId?: string
  creditExpiryDate: string | Date
}

export interface IProductProps {
  productId: Products
  subProductId: SubProducts
}

// TODO: move to komgo types
export interface IDisclosedCreditLineSummary {
  counterpartyStaticId: string
  lowestFee: number
  availabilityCount: number
  appetiteCount: number
  _id: string
}

export interface IExtendedCreditLineRequest extends ICreditLineRequest {
  counterpartyName: string
  companyName: string
}

export interface IDisclosedCreditLineSummaryEnriched extends IDisclosedCreditLineSummary {
  counterpartyName: string
  counterpartyLocation: string
}

export interface IDisclosedCreditLineEnriched extends IDisclosedCreditLine {
  companyName: string
  companyLocation: string
  counterpartyName: string
}

export interface IDisclosedCreditLine extends ICreditLine {
  ownerStaticId: string
  data?: any // TODO: move later to the right type inside `ICreditLine`
}

export interface IRequestInformationForm {
  mailTo: boolean
  comment: string
  requestForId: string
  companyIds: string[]
}

export interface IRequestCreditLineForm extends IRequestInformationForm {
  context: IProductContext
}

export interface IMailToData {
  email?: string
  subject?: string
  body?: string
}

export interface RiskCoverProperties {
  creditLinesById: Map<string, ICreditLineResponse>
  disclosedCreditLineSummariesById: Map<string, IDisclosedCreditLineSummary>
  disclosedCreditLinesById: Map<string, IDisclosedCreditLine>
  requestsById: Map<string, ICreditLineRequest>
}

export interface CreditLinesProperties {
  [CreditLineType.RiskCover]: ImmutableMap<RiskCoverProperties>
  [CreditLineType.BankLine]: ImmutableMap<RiskCoverProperties>
}

export type CreditLinesState = ImmutableMap<CreditLinesProperties>

export interface IMemberWithDisabledFlag extends IMember {
  disabled?: boolean
}
