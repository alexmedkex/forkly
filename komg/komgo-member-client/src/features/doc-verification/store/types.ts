import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'

export enum DocumentVerificationActionType {
  GET_SESSION_REQUEST = '@@licenses/GET_SESSION_REQUEST',
  GET_SESSION_SUCCESS = '@@licenses/GET_SESSION_SUCCESS',
  GET_SESSION_FAILURE = '@@licenses/GET_SESSION_FAILURE',
  VERIFY_DOCUMENT_REQUEST = '@@licenses/VERIFY_DOCUMENT_REQUEST',
  VERIFY_DOCUMENT_SUCCESS = '@@licenses/VERIFY_DOCUMENT_SUCCESS',
  VERIFY_DOCUMENT_FAILURE = '@@licenses/VERIFY_DOCUMENT_FAILURE'
}

export interface DocVerificationStateProperties {
  registeredAt: number
  companyName: string
  metadataHash: string
}

export type DocVerificationState = ImmutableMap<DocVerificationStateProperties>

export interface GetSessionRequest extends Action {
  type: DocumentVerificationActionType.GET_SESSION_REQUEST
}

export interface GetSessionSuccess extends Action {
  type: DocumentVerificationActionType.GET_SESSION_SUCCESS
  payload: { metadataHash: string }
}

export interface GetSessionFailure extends Action {
  type: DocumentVerificationActionType.GET_SESSION_FAILURE
  payload: string
}

export interface VerifyDocumentRequest extends Action {
  type: DocumentVerificationActionType.VERIFY_DOCUMENT_REQUEST
}

export interface VerifyDocumentSuccess extends Action {
  type: DocumentVerificationActionType.VERIFY_DOCUMENT_SUCCESS
  payload: { registeredAt: number; companyName: string }
}

export interface VerifyDocumentFailure extends Action {
  type: DocumentVerificationActionType.VERIFY_DOCUMENT_FAILURE
  payload: string
}

export type DocumentVerificationActions =
  | GetSessionRequest
  | GetSessionSuccess
  | GetSessionFailure
  | VerifyDocumentRequest
  | VerifyDocumentSuccess
  | VerifyDocumentFailure
