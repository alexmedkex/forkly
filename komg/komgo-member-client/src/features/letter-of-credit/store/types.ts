import { Action } from 'redux'
import { IDataLetterOfCredit, ILetterOfCredit, LetterOfCreditTaskType } from '@komgo/types'
import { ImmutableMap } from '../../../utils/types'
import { List, Map } from 'immutable'
import { Task } from '../../tasks/store/types'

export enum LetterOfCreditActionType {
  CREATE_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/CREATE_STANDBY_LETTER_OF_CREDIT_REQUEST',
  CREATE_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/CREATE_STANDBY_LETTER_OF_CREDIT_SUCCESS',
  CREATE_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/CREATE_STANDBY_LETTER_OF_CREDIT_FAILURE',
  FETCH_LETTERS_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/FETCH_LETTERS_OF_CREDIT_REQUEST',
  FETCH_LETTERS_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/FETCH_LETTERS_OF_CREDIT_SUCCESS',
  FETCH_LETTERS_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/FETCH_LETTERS_OF_CREDIT_FAILURE',
  GET_LETTER_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/GET_LETTER_OF_CREDIT_REQUEST',
  GET_LETTER_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/GET_LETTER_OF_CREDIT_SUCCESS',
  GET_LETTER_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/GET_LETTER_OF_CREDIT_FAILURE',
  ISSUE_LETTER_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/ISSUE_LETTER_OF_CREDIT_REQUEST',
  ISSUE_LETTER_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/ISSUE_LETTER_OF_CREDIT_SUCCESS',
  ISSUE_LETTER_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/ISSUE_LETTER_OF_CREDIT_FAILURE',
  REJECT_LETTER_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/REJECT_LETTER_OF_CREDIT_REQUEST',
  REJECT_LETTER_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/REJECT_LETTER_OF_CREDIT_SUCCESS',
  REJECT_LETTER_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/REJECT_LETTER_OF_CREDIT_FAILURE'
}

export type ILetterOfCreditWithData = ILetterOfCredit<IDataLetterOfCredit>

export interface FetchLettersOfCreditSuccessAction extends Action {
  type: LetterOfCreditActionType.FETCH_LETTERS_OF_CREDIT_SUCCESS
  payload: { limit: number; skip: number; total: number; items: ILetterOfCreditWithData[] }
}

export interface GetLetterOfCreditSuccessAction extends Action {
  type: LetterOfCreditActionType.GET_LETTER_OF_CREDIT_SUCCESS
  payload: ILetterOfCreditWithData
}

export interface TemplateStateProperties {
  byStaticId: Map<string, Map<keyof ILetterOfCreditWithData, any>>
  staticIds: List<string>
  total: number
}

export type TemplatedLetterOfCreditState = ImmutableMap<TemplateStateProperties>

export interface ILetterOfCreditEnriched extends ILetterOfCredit<IDataLetterOfCredit> {
  role: string
  actionStatus: string
  latestShipment: string | number | Date
  tasks: Task[]
}

export const LETTER_OF_CREDIT_TYPE_LABELS = {
  [LetterOfCreditTaskType.ReviewIssued]: 'Review issuance',
  [LetterOfCreditTaskType.ReviewRequested]: 'Review request'
}
