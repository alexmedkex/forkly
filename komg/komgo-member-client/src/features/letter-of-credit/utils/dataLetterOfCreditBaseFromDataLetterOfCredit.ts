import { IDataLetterOfCredit, IDataLetterOfCreditBase } from '@komgo/types'

export const buildDataLetterOfCreditBase = (input: IDataLetterOfCredit): IDataLetterOfCreditBase => {
  return {
    version: input.version,
    applicant: {
      staticId: input.applicant.staticId
    },
    issuingBank: {
      staticId: input.issuingBank.staticId
    },
    beneficiary: {
      staticId: input.beneficiary.staticId
    },
    beneficiaryBank: input.beneficiaryBank ? { staticId: input.beneficiaryBank.staticId } : undefined,
    trade: {
      source: input.trade.source,
      sourceId: input.trade.sourceId
    },
    cargo: input.cargo
      ? {
          source: input.cargo.source,
          cargoId: input.cargo.cargoId,
          sourceId: input.cargo.sourceId
        }
      : undefined,
    amount: input.amount,
    expiryDate: input.expiryDate,
    currency: input.currency,
    issuingBankReference: input.issuingBankReference || ''
  }
}
