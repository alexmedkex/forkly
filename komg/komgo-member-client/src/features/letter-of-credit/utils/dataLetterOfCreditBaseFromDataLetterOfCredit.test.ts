import React from 'react'
import { buildDataLetterOfCreditBase } from './dataLetterOfCreditBaseFromDataLetterOfCredit'
import { buildFakeDataLetterOfCredit } from '@komgo/types'

describe('dataLetterOfCreditBaseFromDataLetterOfCredit', () => {
  it('works when cargo is undefined', () => {
    const data = buildFakeDataLetterOfCredit()
    delete data.cargo

    const base = buildDataLetterOfCreditBase(data)

    expect(base).toMatchSnapshot()
  })
})
