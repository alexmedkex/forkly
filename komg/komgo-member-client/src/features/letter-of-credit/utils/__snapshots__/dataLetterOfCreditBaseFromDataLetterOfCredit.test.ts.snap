// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataLetterOfCreditBaseFromDataLetterOfCredit works when cargo is undefined 1`] = `
Object {
  "amount": 10000,
  "applicant": Object {
    "staticId": "cf63c1f8-1165-4c94-a8f8-9252eb4f0016",
  },
  "beneficiary": Object {
    "staticId": "ecc3b179-00bc-499c-a2f9-f8d1cc58e9db",
  },
  "beneficiaryBank": Object {
    "staticId": "08e9f8e3-94e5-459e-8458-ab512bee6e2c",
  },
  "cargo": undefined,
  "currency": "USD",
  "expiryDate": "2019-11-29",
  "issuingBank": Object {
    "staticId": "08e9f8e3-94e5-459e-8458-ab512bee6e2c",
  },
  "issuingBankReference": "REF-0001",
  "trade": Object {
    "source": "KOMGO",
    "sourceId": "E2389423",
  },
  "version": 1,
}
`;
